src/lib.rs

```diff
@@ -227,28 +219,19 @@ pub struct VsockStream {
 
 impl VsockStream {
     /// Open a connection to a remote host.
-    pub fn connect(addr: &SockAddr) -> Result<Self> {
-        let vsock_addr = if let SockAddr::Vsock(addr) = addr {
-            addr.0
-        } else {
+    pub fn connect(addr: &VsockAddr) -> Result<Self> {
+        if addr.family() != Some(AddressFamily::Vsock) {
```

Comment:
I think we don't need this line any more, because the family of VsockAddr must be AddressFamily::Vsock

Reply:
It is necessary now that I have changed the type to `impl SockaddrLike` as requested above.